src/block/BigDripleaf.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\DripleafTiltType;
+use pocketmine\block\utils\HorizontalFacingTrait;
+use pocketmine\block\utils\SupportType;
+use pocketmine\data\runtime\RuntimeDataDescriber;
+use pocketmine\entity\Entity;
+use pocketmine\entity\Living;
+use pocketmine\entity\projectile\Projectile;
+use pocketmine\event\block\StructureGrowEvent;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Facing;
+use pocketmine\math\RayTraceResult;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\utils\AssumptionFailedError;
+use pocketmine\world\BlockTransaction;
+use pocketmine\world\sound\DripleafTiltDownSound;
+use pocketmine\world\sound\DripleafTiltUpSound;
+
+class BigDripleaf extends Transparent{
+	use HorizontalFacingTrait;
+
+	protected DripleafTiltType $tilt;
+	protected bool $head = false;
+
+	public function __construct(BlockIdentifier $idInfo, string $name, BlockTypeInfo $typeInfo){
+		$this->tilt = DripleafTiltType::NONE();
+		parent::__construct($idInfo, $name, $typeInfo);
+	}
+
+	protected function describeBlockOnlyState(RuntimeDataDescriber $w) : void{
+		$w->horizontalFacing($this->facing);
+		$w->dripleafTiltType($this->tilt);
+		$w->bool($this->head);
+	}
+
+	/** @return $this */
+	public function setTilt(DripleafTiltType $tilt) : self{
+		$this->tilt = $tilt;
+		return $this;
+	}
+
+	public function getTilt() : DripleafTiltType{
+		return $this->tilt;
+	}
+
+	/** @return $this */
+	public function setHead(bool $head) : self{
+		$this->head = $head;
+		return $this;
+	}
+
+	public function isHead() : bool{
+		return $this->head;
+	}
+
+	private function canBeSupportedBy(Block $block) : bool{
+		//TODO: Moss block
+		return
+			$block->hasSameTypeId($this) ||
+			$block->getTypeId() === BlockTypeIds::CLAY ||
+			$block->hasTypeTag(BlockTypeTags::DIRT) ||
+			$block->hasTypeTag(BlockTypeTags::MUD);
+	}
+
+	public function onNearbyBlockChange() : void{
+		if(
+			!$this->head && !$this->getSide(Facing::UP)->hasSameTypeId($this) ||
```

Comment:
Potential operator precedence issue here, I assume the below is the correct solution?
```suggestion
			(!$this->head && !$this->getSide(Facing::UP)->hasSameTypeId($this)) ||
```